Bump arsenal and update supportedLifecycleRules

francoisferrand opened the pull request:
- Bump arsenal 8.2.18
- Upgrade supportedLifecycleRules definition

Issue: BB-688

francoisferrand committed:
Upgrade supportedLifecycleRules definition

Use the values and "format" defined in arsenal, i.e. the names from S3.
This simplifies code and avoids duplication.

Issue: BB-688

francoisferrand committed:
Bump arsenal 8.2.18

Issye: BB-688

francoisferrand committed:
Release backbeat 9.0.8

Issue: BB-688

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

williamlardier commented on tests/functional/queuePopulator/config/s3c-config.json:
```diff
@@ -141,9 +141,9 @@
-        "abortIncompleteMultipartUpload"
+        "Expiration",
+        "NoncurrentVersionExpiration",
+        "AbortIncompleteMultipartUpload"
     ]
     }
```
the (untouched) indentation looks wrong in this file

williamlardier commented on extensions/lifecycle/tasks/LifecycleTask.js:
```diff
@@ -95,11 +95,7 @@ class LifecycleTask extends BackbeatTask {
-            this._lifecycleDateTime
-        );
-        this._supportedRules = this.supportedRules;
+        this.setSupportedRules(this.supportedRules);
```
We should test this in some unit test to fix the codecov warning? (same for other places)

williamlardier commented on tests/functional/lifecycle/LifecycleTaskV2-versioned.js:
```diff
@@ -83,13 +85,7 @@ describe('LifecycleTaskV2 with bucket versioned', () => {
-                    'transitions',
-                    'noncurrentVersionTransition'
-                ],
+                supportedRules: ValidLifecycleRules
```
```suggestion
                supportedRules: ValidLifecycleRules,
```

francoisferrand committed:
Update bucketMD with arsenal 8.2

Some fields used to default to `null`, are not set to undefined, see [1]

[1] https://github.com/scality/Arsenal/pull/2299

Issue: BB-688

francoisferrand committed:
Fix zk cleanup when lifecycle config deleted

Issue: BB-688

francoisferrand commented on package.json:
```diff
@@ -47,7 +47,7 @@
   "dependencies": {
     "JSONStream": "^1.3.5",
-    "arsenal": "git+https://github.com/scality/arsenal#8.2.2",
+    "arsenal": "git+https://github.com/scality/arsenal#e30af07b745e861bdcf05a43f077ad0c54c3cbeb",
```
will be updated before merging (once arsenal available)

francoisferrand commented:
/approve

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/9.1/improvement/BB-688` with contents from `improvement/BB-688`
and `development/9.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/9.1/improvement/BB-688 origin/development/9.1
 $ git merge origin/improvement/BB-688
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/9.1/improvement/BB-688
```





*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/9.0`

* :heavy_check_mark: `development/9.1`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`



Please check the status of the associated issue BB-688.

Goodbye francoisferrand.




*The following options are set:* **approve**
